date_range/wizard/date_range_generator.py

```diff
@@ -0,0 +1,81 @@
+# Copyright 2016 ACSONE SA/NV (<http://acsone.eu>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+from dateutil.relativedelta import relativedelta
+from dateutil.rrule import DAILY, MONTHLY, WEEKLY, YEARLY, rrule
+
+from odoo import _, api, fields, models
+from odoo.exceptions import ValidationError
+
+
+class DateRangeGenerator(models.TransientModel):
+    _name = 'date.range.generator'
+    _description = 'Date Range Generator'
+
+    @api.model
+    def _default_company(self):
+        return self.env['res.company']._company_default_get('date.range')
```

Comment:
Shouldn't be:

```suggestion
        return self.env.company
```

?